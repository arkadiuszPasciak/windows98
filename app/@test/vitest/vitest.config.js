import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
