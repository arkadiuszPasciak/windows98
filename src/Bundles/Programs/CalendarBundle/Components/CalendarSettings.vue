<template>
  <UIFrame class="CalendarSettings" :title="t('CalendarBundle.title')">
    <CalendarMonthSwitcher />
    <CalendarYearSwitcher />
    <CalendarView />
  </UIFrame>
</template>

<script setup lang="ts">
  import { useI18n } from 'vue-i18n'
  import CalendarMonthSwitcher from '@Bundles/CalendarBundle/Components/CalendarMonthSwitcher.vue'
  import CalendarYearSwitcher from '@Bundles/CalendarBundle/Components/CalendarYearSwitcher.vue'
  import CalendarView from '@Bundles/CalendarBundle/Components/CalendarView.vue'
  import UIFrame from '@Bundles/UIBundle/Components/UIFrame.vue'

  const { t } = useI18n()
</script>

<i18n src="@Bundles/CalendarBundle/Locales/Calendar.locales.json"></i18n>

<style
  lang="scss"
  scoped
  src="@Bundles/CalendarBundle/Styles/CalendarSettings.styles.scss"
/>
