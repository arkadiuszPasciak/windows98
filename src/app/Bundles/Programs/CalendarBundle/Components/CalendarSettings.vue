<template>
  <UIFrame class="CalendarSettings" :title="t('CalendarBundle.title')">
    <CalendarMonthSwitcher />
    <CalendarYearSwitcher />
    <CalendarView />
  </UIFrame>
</template>

<script setup lang="ts">
  import { useI18n } from 'vue-i18n'
  import CalendarMonthSwitcher from '@APP|Bundles/CalendarBundle/Components/CalendarMonthSwitcher.vue'
  import CalendarYearSwitcher from '@APP|Bundles/CalendarBundle/Components/CalendarYearSwitcher.vue'
  import CalendarView from '@APP|Bundles/CalendarBundle/Components/CalendarView.vue'
  import UIFrame from '@APP|Bundles/UIFrameBundle/Components/UIFrame.vue'

  const { t } = useI18n()
</script>

<i18n src="@APP|Bundles/CalendarBundle/Locales/Calendar.locales.json"></i18n>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/CalendarBundle/Styles/CalendarSettings.styles.scss"
/>
