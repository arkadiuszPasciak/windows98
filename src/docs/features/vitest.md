# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
