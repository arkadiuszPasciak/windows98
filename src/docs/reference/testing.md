# Testing

This documentation provides an overview of the testing process employed in our application. We utilize two powerful testing tools to ensure the reliability and stability of the code: Cypress and Vitest. These tools are employed in different aspects of testing to maximize coverage and maintain code quality.

## [Vitest](https://vitest.dev/)

`@/vite.config.ts` (in the block test)

Vitest is a lightweight and fast testing framework specifically designed for Vite applications. It provides seamless integration with Vite, supporting both JavaScript and TypeScript code. Vitest is particularly well-suited for unit testing, which focuses on verifying the functionality of individual code units, such as functions, classes, or modules.

We use Vitest for `unit testing` our TypeScript code, ensuring that each function or class in our application performs as intended. This approach allows us to isolate potential issues and fix them before they cause problems in the broader application context.

To effectively use Vitest, we create a series of test suites that group related tests, with each test case focusing on a specific unit of code. These test cases cover various scenarios, such as input validation, error handling, and edge cases, to ensure comprehensive coverage of our codebase.
