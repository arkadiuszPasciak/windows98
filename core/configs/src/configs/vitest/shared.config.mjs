import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
