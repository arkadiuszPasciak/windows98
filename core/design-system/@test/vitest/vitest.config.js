import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"

export default mergeConfig(
	sharedConfig,
	defineProject({/* here custom configs */ })
)
