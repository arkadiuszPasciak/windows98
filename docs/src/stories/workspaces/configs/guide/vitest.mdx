import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

