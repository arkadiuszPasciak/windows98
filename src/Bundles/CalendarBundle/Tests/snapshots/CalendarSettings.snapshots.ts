import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import { createTestingPinia } from '@pinia/testing'
import CalendarSettings from '@Bundles/CalendarBundle/Components/CalendarSettings.vue'

const defaultWrapper = mount(CalendarSettings, {
  global: {
    plugins: [createTestingPinia()],
  },
})

describe('[CalendarBundle]<Snapshots>(CalendarSettings)', async () => {
  it('renders correctly component', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
})
