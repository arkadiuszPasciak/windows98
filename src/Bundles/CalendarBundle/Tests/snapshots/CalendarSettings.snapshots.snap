// Vitest Snapshot v1

exports[`[CalendarBundle]<Snapshots>(CalendarSettings) > renders correctly component 1`] = `
<div
  class="UIFrame CalendarSettings"
  data-v-9351c3bb=""
  data-v-ff9b8841=""
>
  <p
    class="UIText title"
    data-v-f0b878d5=""
    data-v-ff9b8841=""
  >
    
    Calendar
    
  </p>
  
  <select
    class="UISelect variant-undefined CalendarMonthSwitcher"
    data-v-0ce71445=""
    data-v-9351c3bb=""
    data-v-d3c68ad1=""
    id="date-months"
    name="date-months"
  >
    
    <option
      class="option"
      data-v-d3c68ad1=""
      value="0"
    >
      January
    </option>
    <option
      class="option"
      data-v-d3c68ad1=""
      value="1"
    >
      February
    </option>
    <option
      class="option"
      data-v-d3c68ad1=""
      value="2"
    >
      March
    </option>
    <option
      class="option"
      data-v-d3c68ad1=""
      value="3"
    >
      April
    </option>
    <option
      class="option"
      data-v-d3c68ad1=""
      value="4"
    >
      May
    </option>
    <option
      class="option"
      data-v-d3c68ad1=""
      value="5"
    >
      June
    </option>
    <option
      class="option"
      data-v-d3c68ad1=""
      value="6"
    >
      July
    </option>
    <option
      class="option"
      data-v-d3c68ad1=""
      value="7"
    >
      August
    </option>
    <option
      class="option"
      data-v-d3c68ad1=""
      value="8"
    >
      September
    </option>
    <option
      class="option"
      data-v-d3c68ad1=""
      value="9"
    >
      October
    </option>
    <option
      class="option"
      data-v-d3c68ad1=""
      value="10"
    >
      November
    </option>
    <option
      class="option"
      data-v-d3c68ad1=""
      value="11"
    >
      December
    </option>
    
  </select>
  <div
    class="CalendarYearSwitcher"
    data-v-9351c3bb=""
    data-v-ca14cfec=""
  >
    <p
      class="UIText year"
      data-v-ca14cfec=""
      data-v-f0b878d5=""
    >
      
      
      
    </p>
    <button
      class="UIButton size-small button increase"
      data-v-ca14cfec=""
      data-v-cf1bb3a9=""
      type="button"
    >
      
      
      
      
      
      
    </button>
    <button
      class="UIButton size-small button decrease"
      data-v-ca14cfec=""
      data-v-cf1bb3a9=""
      type="button"
    >
      
      
      
      
      
      
    </button>
  </div>
  <div
    class="CalendarView"
    data-v-4334b8a6=""
    data-v-9351c3bb=""
  >
    <div
      class="week-days"
      data-v-4334b8a6=""
    >
      
      <div
        class="week"
        data-v-4334b8a6=""
      >
        S
      </div>
      <div
        class="week"
        data-v-4334b8a6=""
      >
        M
      </div>
      <div
        class="week"
        data-v-4334b8a6=""
      >
        T
      </div>
      <div
        class="week"
        data-v-4334b8a6=""
      >
        W
      </div>
      <div
        class="week"
        data-v-4334b8a6=""
      >
        T
      </div>
      <div
        class="week"
        data-v-4334b8a6=""
      >
        F
      </div>
      <div
        class="week"
        data-v-4334b8a6=""
      >
        S
      </div>
      
    </div>
    <!--v-if-->
  </div>
  
</div>
`;
